dev/lsp/src/features/completion.ts

```diff
@@ -1,13 +1,65 @@
-import { TextDocumentPositionParams } from "vscode-languageserver";
+import {
+  TextDocumentPositionParams,
+  CompletionList,
+  CompletionItem,
+  CompletionItemKind,
+  Range,
+  Position,
+  CompletionItemLabelDetails,
+} from "vscode-languageserver";
+import { URI } from "vscode-uri";
 import { LSPInstance } from "../lsp";
+import { 
+    getBuiltinsDoc, 
+    LobsterBuiltinsDoc, 
+    LobsterFunctionSignature 
+} from "../lobster";
+
+let builtinsDoc: LobsterBuiltinsDoc[];
+
+export function getFunctionSignature(
+  funcname: string,
+): LobsterFunctionSignature | undefined {
+  const doc: LobsterBuiltinsDoc = builtinsDoc.find((d) => d.funcname == funcname)!;
+  const res: LobsterFunctionSignature = {
+    name: doc.funcname,
+    parameters: doc.args,
+  };
+  return res;
+}
 
 export default function setupFeature(lsp: LSPInstance) {
-    lsp.connection.onCompletion(
-        async (_textDocumentPosition: TextDocumentPositionParams) => {
-            if (lsp.errored()) return [];
-            
-            // TODO
-            return [];
-        }
-    );
-}
\ No newline at end of file
+  lsp.connection.onCompletion(
+    async (_textDocumentPosition: TextDocumentPositionParams) => {
+      if (!builtinsDoc) {
+        const settings = await lsp.getDocumentSettings(URI.parse(_textDocumentPosition.textDocument.uri));  
+        builtinsDoc = await getBuiltinsDoc(settings);
```

Comment:
Same as above shouldn't be in this feature, should prob be stored in LSPInstance